src/sqlfluff/core/templaters/jinja.py

```diff
@@ -393,7 +393,15 @@ def __getattr__(self, item):
 
         for val in potentially_undefined_variables:
             if val not in live_context:
-                live_context[val] = Undefined(name=val)
+                if not config or "templating" not in config.get("ignore"):
+                    live_context[val] = Undefined(name=val)
+                else:
+                    # When ignoring=templating is configured, provide a
+                    # "reasonable" default for undefined variables, rather than
+                    # trying to force them to fail. This won't always work
+                    # (i.e. there's no "universal default" value that will) work
+                    # in all cases, but this will probably work 90% of the time.
+                    live_context[val] = "a"
```

Comment:
Should we emit an error or warning here to make it clear we're doing this? Even if it's just in debug mode.

Reply:
Sure, I'll add that.

Reply:
Done!